src/core/hle/service/ir/ir_rst.cpp

```diff
@@ -51,34 +50,22 @@ struct SharedMem {
 
 static_assert(sizeof(SharedMem) == 0x98, "SharedMem has wrong size!");
 
-void IR_RST::LoadInputDevices() {
-    zl_button = Input::CreateDevice<Input::ButtonDevice>(
-        Settings::values.current_input_profile.buttons[Settings::NativeButton::ZL]);
-    zr_button = Input::CreateDevice<Input::ButtonDevice>(
-        Settings::values.current_input_profile.buttons[Settings::NativeButton::ZR]);
-    c_stick = Input::CreateDevice<Input::AnalogDevice>(
-        Settings::values.current_input_profile.analogs[Settings::NativeAnalog::CStick]);
-}
-
 void IR_RST::UnloadInputDevices() {
```

Comment:
You don't need to unload anything from core.  Only the front end should finalize input